lib/src/subscription.dart

```diff
@@ -0,0 +1,284 @@
+/// block.
+///
+/// Any changes to the subscription set will be persisted locally and be available the next
+/// time the application starts up - i.e. it's not necessary to subscribe for the same query
+/// every time. Updating the subscription set can be done while offline, and only the latest
+/// update will be sent to the server whenever connectivity is restored.
+///
+/// It is strongly recommended that you batch updates as much as possible and request the
+/// dataset your application needs upfront. Updating the set of active subscriptions for a
+/// Realm is an expensive operation serverside, even if there's very little data that needs
+/// downloading.
+/// {@category Sync}
+abstract class SubscriptionSet with IterableMixin<Subscription> {
+  Realm _realm;
+  SubscriptionSetHandle _handle;
+
+  SubscriptionSet._(this._realm, this._handle);
+
+  /// Finds an existing [Subscription] in this set by its query
+  ///
+  /// The [query] is represented by the corresponding [RealmResults] object.
+  /// Finds a subscription by query.
+  ///
+  /// If the Subscription set does not contain a subscription with the provided query,
+  /// return null
+  Subscription? find<T extends RealmObject>(RealmResults<T> query) {
+    return realmCore.findSubscriptionByQuery(this, query).convert(Subscription._);
+  }
+
+  /// Finds an existing [Subscription] in this set by name.
+  ///
+  /// If the Subscription set does not contain a subscription with the provided name,
+  /// return null
+  Subscription? findByName(String name) {
+    return realmCore.findSubscriptionByName(this, name).convert(Subscription._);
+  }
+
+  Future<SubscriptionSetState> _waitForStateChange(SubscriptionSetState state) async {
+    final result = await realmCore.waitForSubscriptionSetStateChange(this, state);
+    realmCore.refreshSubscriptionSet(this);
+    return result;
+  }
+
+  /// Waits for the server to acknowledge the subscription set and return the matching objects.
+  ///
+  /// If the [state] of the subscription set is [SubscriptionSetState.complete]
+  /// the returned [Future] will complete immediately. If the state is
+  /// [SubscriptionSetState.error], the returned future will throw an
+  /// error.
+  Future<void> waitForSynchronization() async {
+    final result = await _waitForStateChange(SubscriptionSetState.complete);
+    if (result == SubscriptionSetState.error) {
+      throw RealmException('Synchronization Failed');
+    }
+  }
+
+  @override
+  int get length => realmCore.getSubscriptionSetSize(this);
+
+  @override
+  Subscription elementAt(int index) {
+    RangeError.checkValidRange(index, null, length);
+    return Subscription._(realmCore.subscriptionAt(this, index));
+  }
+
+  /// Gets the [Subscription] at the specified index in the set.
+  Subscription operator [](int index) => elementAt(index);
+
+  @override
+  _SubscriptionIterator get iterator => _SubscriptionIterator._(this);
```

Comment:
No, it is inherited. It will inherit the docs as well.
<img width="705" alt="image" src="https://user-images.githubusercontent.com/22237677/168537030-d255baaf-d876-472e-8631-5b433bcec8ce.png">